### PR TITLE
yii\rest\Serializer serialize arrays

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️?
| Fixed issues  | #17886

### CODE REVIEW

**Comment by samdark** [comment]
@patacca could you add some unit tests to ensure this behavior won't be broken in the future?

**Comment by patacca** [comment]
I am gonna try to do it tomorrow

**Comment by samdark** [comment]
@patacca did https://github.com/yiisoft/yii2/pull/17888 fix the issue?

**Comment by patacca** [comment]
No because php array is not an instance of JsonSerializable

**Comment by patacca** [comment]
I've made the unit test and merged the branch with yiisoft/yii2 master

File: framework/rest/Serializer.php [hunk 1]
```diff
@@ -152,6 +152,11 @@ public function serialize($data)
             return $this->serializeModel($data);
         } elseif ($data instanceof DataProviderInterface) {
             return $this->serializeDataProvider($data);
+        } elseif (is_array($data)) {
+            $serializedArray = [];
+            foreach ($data as $key => $value)
```
Comment (samdark) [inline framework/rest/Serializer.php]: `{` and `}` should be added for foreach.

**Review by samdark — COMMENTED** [review]
All good. Please fix formatting and add a line for changelog.

File: framework/rest/Serializer.php (hunk 1 above)
Comment (patacca) [reply framework/rest/Serializer.php]: I've done both. You can find everything in this branch

**Comment by samdark** [comment]
@patacca merged in https://github.com/yiisoft/yii2/commit/ecbafa2e3aad55d83afee190ada41bf42e00f131. Thanks!

**Pull request closed without merging**